src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.BetDaoImpl;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.dao.UserDaoImpl;
+import core.basesyntax.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+
+    public static Object getInstance(Class clas) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException,
+            InstantiationException, NoClassAnnotation {
+        Constructor constructor = clas.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Field[] declaredFields = clas.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                field.setAccessible(true);
+                if (checkBet(field)) {
+                    field.set(instance, Factory.getBetDao());
+                    continue;
+                }
+                if (checkUser(field)) {
+                    field.set(instance, Factory.getUserDao());
+                }
```

Comment:
let's refactor lines 23-29

1. check if the field's type is BetDao and there is `@Dao` in BetDaoImpl - then set field
2. else - check if the field's type is UserDao and there is `@Dao` in UserDaoImpl - then set field
3. else - throw your custom Exception

Reply:
fixed